### PR TITLE
bump RTD to py39

### PR DESCRIPTION
**Pull request by keflavich**
Attempt to fix https://github.com/astropy/astroquery/pull/2729 outstanding CI failure

### CODE REVIEW

**Comment by keflavich** [comment]
"Problem in your project's configuration. Invalid "python.version": expected one of (2, 2.7, 3, 3.5, 3.6, 3.7, 3.8, pypy3.5), got 3.9"

🤦

**Comment by keflavich** [comment]
https://docs.readthedocs.io/en/stable/config-file/v2.html#build-beta-specification apparently we were using an old-style (deprecated) version specification

File: astroquery/mast/tests/test_mast.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ def test_mast_service_request(patch_post):
 def test_resolve_object(patch_post):
     m103_loc = mast.Mast.resolve_object("M103")
     print(m103_loc)
-    assert m103_loc.separation(SkyCoord("23.34086 60.658", unit='deg')).value == 0
+    assert round(m103_loc.separation(SkyCoord("23.34086 60.658", unit='deg')).value, 10) == 0
```
Comment (bsipocz) [inline astroquery/mast/tests/test_mast.py:305]: Why was this change included in this RTD config PR?

File: .readthedocs.yaml [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
-       path: .
-       extra_requirements:
-          - docs
+build:
+  os: "ubuntu-20.04"
+  tools:
+    python: "mambaforge-4.10"
+
+conda:
+  environment: docs/rtd_environment.yaml
```
Comment (bsipocz) [inline .readthedocs.yaml:9]: Would be nice to not rely on any conda stuff for CI. Do you plan to have this temporarily, or this is now our solution for docs builds?

File: astroquery/mast/tests/test_mast.py (hunk 1 above)
Comment (keflavich) [reply astroquery/mast/tests/test_mast.py:305]: These were simultaneous fixes - #2729 depended on #2730 but both were generically "fix CI" PRs.

File: .readthedocs.yaml (hunk 2 above)
Comment (keflavich) [reply .readthedocs.yaml:9]: Great question.  I did this because it seemed the most direct way to get RTD to build again.  I'm totally happy to return to a non-conda approach.  I was unable to figure one out - graphviz specifically caused problems.

Comment (bsipocz) [reply .readthedocs.yaml:9]: And did you find out why it stopped working the first place? As grapviz wasn't really causing issues recently

Comment (keflavich) [reply .readthedocs.yaml:9]: This error started appearing spontaneously:
https://readthedocs.org/projects/astroquery/builds/20444648/
"Could not import extension sphinx.builders.linkcheck (exception: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with OpenSSL 1.0.2n  7 Dec 2017. See: https://github.com/urllib3/urllib3/issues/2168)"

My best read was that there is a problem caused by using old python3.8.  We then had to upgrade to the new RTD configuration system, and then _that_ caused problems with graphviz until I switched to the astropy conda-based version.

Comment (bsipocz) [reply .readthedocs.yaml:9]: yeap, but graphviz can be installed into the image directly without adding conda into the picture. #2732 should fix it.
